File: csharp/binary/Binary.cs
using System;

public class Binary
{
  public static int ToDecimal(string binary)
  {
    int output = 0, i = 0;
    for (int len = binary.Length - 1; len >= 0; len--)
    {
      if (!ValidInput(binary[i]))
      {
        return 0;
      }
      output += Int32.Parse(binary[i].ToString()) * (int)Math.Pow(2, len);
      i++;
    }
    return output;
  }

  private static bool ValidInput(char input)
  {
    return input == '0' || input == '1';
  }
}

File: javascript/binary/binary.js
function Binary(input) {
  this.toDecimal = function() {
    return input.match(/[^01]/g) ? 0 : input.split("").reverse().reduce((p, c, i) => p + c * Math.pow(2, i), 0);
  };
};

module.exports = Binary;
